test/integration-test/src/tests/btf_relocations.rs

```diff
@@ -163,47 +183,86 @@ struct RelocationTest {
 
 impl RelocationTest {
     /// Build a RelocationTestRunner
-    fn build(&self) -> Result<RelocationTestRunner> {
+    fn build(&self, relocation_code_in_prog: bool) -> Result<RelocationTestRunner> {
         Ok(RelocationTestRunner {
-            ebpf: self.build_ebpf()?,
+            ebpf: self.build_ebpf(relocation_code_in_prog)?,
             btf: self.build_btf()?,
         })
     }
 
     /// - Generate the source eBPF filling a template
     /// - Compile it with clang
-    fn build_ebpf(&self) -> Result<Vec<u8>> {
+    fn build_ebpf(&self, relocation_code_in_prog: bool) -> Result<Vec<u8>> {
```

Comment:
I think this is probably unnecessary? We could always wrap in a func, relocating
at the toplevel is a subset of relocating nested calls afterall, so if the
latter works it's unlikely that the former is going to silently break.